### ISSUE TITLE
Generate docker-compose file using Docker API

### ISSUE THREAD

**Issue by egeldenhuys** [TripleParity/docks-api#61]
Given a stack name it should be possible to generate a working `docker-compose.yml` file.

This may require first discovering all services that are part of the given stack, and reverse each one of them individually. I think Swarmpit already does this, so we can have a look at how they are doing it.

If we can reverse a service into a model we can also use this in the UI for populating forms.

- [ ] Describe standard model for representing a service. This can be used in the UI as well
- [ ] Populate model based on `docker inspect` output
- [ ] Convert model to docker-compose section
- [ ] Describe standard model for representing a stack
- [ ] Populate stack model
- [ ] Convert stack model to docker-compose file

We do not need to convert from stack file to the model, since we will give the stack file directly to the Docker API.

I made this a separate issue as it is probably the most complex part of the system so far

**Comment by egeldenhuys** [comment]
Most relevant files from Swarmpit that I could find. @devosray are you familiar with Clojure?

- https://github.com/swarmpit/swarmpit/blob/083ce82dc293d98c201e551e89f4b77f541d35e0/src/clj/swarmpit/docker/engine/mapper/compose.clj
- https://github.com/swarmpit/swarmpit/blob/712b96b008d2d834d0e65ec986cbb0ce382c1e9b/src/cljs/swarmpit/component/stack/edit.cljs
- https://github.com/swarmpit/swarmpit/blob/712b96b008d2d834d0e65ec986cbb0ce382c1e9b/src/cljs/swarmpit/component/service/edit.cljs

Basically they map Docker inspect output to an intermediate model

**Comment by egeldenhuys** [comment]
Some useful tools:
- https://github.com/apiaryio/fury-cli
- https://github.com/bcherny/json-schema-to-typescript

Note that I have not done any further work on this issue

**Comment by egeldenhuys** [comment]
I found a python version that works for containers: https://github.com/Red5d/docker-autocompose

**Comment by devosray** [comment]
Could you maybe explain the relevance of fury-cli? Could you maybe explain how it helps with this task?

**Comment by egeldenhuys** [comment]
Fury converts the `swagger.yml` to a `blueprint.apib` which contains JSON Schemas.
The Schema can then be converted into a TypeScript interface (and probably a relevant structure for JavaScript). I guess this is more relevant to TypeScript since in JavaScript we welcome run time errors

**Comment by egeldenhuys** [comment]
npm package that might be of interest:
- [docker-composer on npm](https://www.npmjs.com/package/docker-composer)
- [docker-composer on GitHub](https://github.com/tudvari/docker-composer)

I think it will be better to fork this project (if we need to modify it) than to create our own package

**Comment by egeldenhuys** [comment]
The npm package repo: https://github.com/TripleParity/docker-api-to-compose
